### PR TITLE
Fix Palette parsing and include "python"

### PR DESCRIPTION
**Pull request by jedbrown**
Fix https://github.com/flamegraph-rs/flamegraph/issues/267

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -518,10 +522,10 @@ pub struct FlamegraphOptions {
     /// Color palette
     #[clap(
         long,
-        value_parser([
+        value_parser = PossibleValuesParser::new([
             "hot", "mem", "io", "red", "green", "blue", "aqua", "yellow",
             "purple", "orange", "wakeup", "java", "perl", "js", "rust"
-        ])
+        ]).map(|s| Palette::from_str(&s).unwrap())
```
Comment (djc) [inline src/lib.rs]: What does the command-line output look like when this panics? Is there a way to yield an error, instead?

Comment (jedbrown) [reply src/lib.rs]: Output is like this, but colored. Exit code 2.
```console
$ cargo run --bin=flamegraph -- --palette pythonx -- true
    Finished dev [unoptimized + debuginfo] target(s) in 0.04s
     Running `target/debug/flamegraph --palette pythonx -- true`
error: invalid value 'pythonx' for '--palette <PALETTE>'
  [possible values: aqua, blue, green, hot, io, java, js, mem, orange, perl, python, purple, red, rust, wakeup, yellow]

  tip: a similar value exists: 'python'

For more information, try '--help'.
```

Comment (jedbrown) [reply src/lib.rs]: Surprisingly, if I remove the `.unwrap()`, it compiles cleanly and gives the same output for an incorrect palette name, but gives this for a correct palette name.
```console
$ RUST_BACKTRACE=1 cargo run --bin=flamegraph -- --palette hot -- true
    Finished dev [unoptimized + debuginfo] target(s) in 0.04s
     Running `target/debug/flamegraph --palette hot -- true`
thread 'main' panicked at 'Mismatch between definition and access of `palette`. Could not downcast to inferno::flamegraph::color::Palette, need to downcast to core::result::Result<inferno::flamegraph::color::Palette, alloc::string::String>
', src/lib.rs:527:18
stack backtrace:
   0: rust_begin_unwind
             at /rustc/fe7454bf439c93cbe9ac8a8f7fcfacd5a40244c2/library/std/src/panicking.rs:593:5
   1: core::panicking::panic_fmt
             at /rustc/fe7454bf439c93cbe9ac8a8f7fcfacd5a40244c2/library/core/src/panicking.rs:67:14
   2: clap_builder::parser::error::MatchesError::unwrap
             at /home/jed/.cargo/registry/src/index.crates.io-6f17d22bba15001f/clap_builder-4.2.7/src/parser/error.rs:32:9
   3: clap_builder::parser::matches::arg_matches::ArgMatches::remove_one
             at /home/jed/.cargo/registry/src/index.crates.io-6f17d22bba15001f/clap_builder-4.2.7/src/parser/matches/arg_matches.rs:409:9
   4: <flamegraph::FlamegraphOptions as clap_builder::derive::FromArgMatches>::from_arg_matches_mut
             at ./src/lib.rs:496:17
   5: <flamegraph::Options as clap_builder::derive::FromArgMatches>::from_arg_matches_mut
             at ./src/lib.rs:466:12
   6: <flamegraph::Opt as clap_builder::derive::FromArgMatches>::from_arg_matches_mut
             at ./src/bin/flamegraph.rs:20:12
   7: clap_builder::derive::Parser::parse
             at /home/jed/.cargo/registry/src/index.crates.io-6f17d22bba15001f/clap_builder-4.2.7/src/derive.rs:28:19
   8: flamegraph::main
             at ./src/bin/flamegraph.rs:31:15
   9: core::ops::function::FnOnce::call_once
             at /rustc/fe7454bf439c93cbe9ac8a8f7fcfacd5a40244c2/library/core/src/ops/function.rs:250:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```
Similar run-time panic if passed through `.ok()`. Perhaps clap could do some `FromResidual` tricks to accept those, but this is how it's meant to be used for now.

**Comment by djc** [comment]
Thanks! Don't like where the clap API is going...

**Comment by jedbrown** [comment]
Thanks. FWIW, the user input is validated against the listed variants so the only way `from_str` fails is if it fails to parse the variants it defines (a coding bug). Or perhaps if instantiating a "`Palette`" needs to acquire a fallible resource like a socket, but in most/all cases, that should be done later.

**Pull request merged**